pkg/controller/node/nodecontroller.go

```diff
@@ -889,6 +1020,32 @@ func (nc *NodeController) evictPods(node *api.Node) bool {
 	return nc.zonePodEvictor[utilnode.GetZoneKey(node)].Add(node.Name, string(node.UID))
 }
 
+// cancelQueuedPodsEviction ensures pod not queued for eviction, typically because
+// the pods now tolerate all NoExecute taints on the node.
+func (nc *NodeController) cancelQueuedPodsEviction(node *api.Node, pods ...api.Pod) {
```

Comment:
It should be pretty much identical to `cancelPodEviction` function. I.e. you need to remove it from both "intolerable" and "termination" evictors and it has to be done under the 'evictorLock'.

You also need to update cancelPodEviction, as you'll be changing semantics of terminationEvictor. Or we can wait until #35235 is merged, and then `cancelPodEviction` should be fine (you'll need to rebase this PR though).